src/main/java/uk/gov/hmcts/reform/draftstore/endpoint/v3/DraftController.java

```diff
@@ -98,8 +104,10 @@ public DraftList readAll(
         @RequestHeader(name = SECRET_HEADER, required = false) @Valid @Length(min = MIN_SECRET_LENGTH) String secretHeader,
         @RequestBody @Valid CreateDraft newDraft
     ) {
-        UserAndService userAndService = authService.authenticate(authHeader, serviceHeader, secretHeader);
-        int id = draftService.create(newDraft, userAndService);
+        Secrets secrets = Secrets.fromHeader(secretHeader);
```

Comment:
We're repeating this sequence a few times:

- get secrets from header
- authenticate using auth header
- add secrets to UserAndService
- perform an action using UserAndService

I'm wondering if we could extract some common code. What do you think?

Reply:
Once we start using Spring Security this will be gone :)